modules/id5IdSystem.js

```diff
@@ -177,83 +242,101 @@ class IdFetchFlow {
     this.usPrivacyData = usPrivacyData
   }
 
-  execute() {
-    return this.#callForConfig(this.submoduleConfig)
-      .then(fetchFlowConfig => {
-        return this.#callForExtensions(fetchFlowConfig.extensionsCall)
-          .then(extensionsData => {
-            return this.#callId5Fetch(fetchFlowConfig.fetchCall, extensionsData)
-          })
-      })
-      .then(fetchCallResponse => {
-        try {
-          resetNb(this.submoduleConfig.params.partner);
-          if (fetchCallResponse.privacy) {
-            storeInLocalStorage(ID5_PRIVACY_STORAGE_NAME, JSON.stringify(fetchCallResponse.privacy), NB_EXP_DAYS);
-          }
-        } catch (error) {
-          logError(LOG_PREFIX + error);
-        }
-        return fetchCallResponse;
-      })
+  /**
+   * Calls the ID5 Servers to fetch an ID5 ID
+   * @returns {Promise<IdResponse>} The result of calling the server side
+   */
+  async execute() {
+    if (this.#isExternalModule()) {
+      try {
+        return this.#externalModuleFlow();
+      } catch (error) {
+        logError('Error while performing ID5 external module flow. Continuing with regular flow.', error);
+        return this.#regularFlow();
+      }
+    } else {
+      return this.#regularFlow();
+    }
   }
 
-  #ajaxPromise(url, data, options) {
-    return new Promise((resolve, reject) => {
-      ajax(url,
-        {
-          success: function (res) {
-            resolve(res)
-          },
-          error: function (err) {
-            reject(err)
-          }
-        }, data, options)
-    })
+  #isExternalModule() {
+    return typeof this.submoduleConfig.params.externalModuleUrl === 'string';
   }
 
   // eslint-disable-next-line no-dupe-class-members
```

Comment:
What's the reason for disabling these linter checks?

Reply:
The reason is beacuse eslint treat each [`#` prefixed (private)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties) method as duplicated . It can't differentiate them even if  they have different names

Reply:
My colleague, who introduced this, said that lint has or at least had issue with # prefixed methods: it recognized them as non-uniqe/duplicated. I'll check again if that's the case.

Reply:
You are right! I was surprised because we use private method elsewhere, but by chance never more than one per class.